eth-contracts/contracts/DelegateManager.sol

```diff
@@ -700,41 +704,54 @@ contract DelegateManager is InitializableV2 {
                and do validation
      * @param spFactory - reference to ServiceProviderFactory contract
-     * @return (totalBalanceInStaking, totalBalanceInSPFactory, totalBalanceOutsideStaking)
+     * @return (totalBalanceInStaking, totalBalanceInSPFactory, totalBalanceOutsideStaking, spLockedStake, totalRewards)
      */
     function _validateClaimRewards(ServiceProviderFactory spFactory)
-    internal returns (uint totalBalanceInStaking, uint totalBalanceInSPFactory, uint totalBalanceOutsideStaking)
-        {
-
+    internal returns (
+        uint totalBalanceInStaking,
+        uint totalBalanceInSPFactory,
+        uint totalBalanceOutsideStaking,
+        uint spLockedStake,
+        uint totalRewards
+    )
+    {
         // Account for any pending locked up stake for the service provider
-        (uint spLockedStake,) = spFactory.getPendingDecreaseStakeRequest(msg.sender);
+        (spLockedStake,) = spFactory.getPendingDecreaseStakeRequest(msg.sender);
+        uint totalLockedUpStake = spDelegateInfo[msg.sender].totalLockedUpStake.add(spLockedStake);
 
         // Process claim for msg.sender
         // Total locked parameter is equal to delegate locked up stake + service provider locked up stake
-        ClaimsManager(claimsManagerAddress).processClaim(
+        uint mintedRewards = ClaimsManager(claimsManagerAddress).processClaim(
             msg.sender,
-            (spDelegateInfo[msg.sender].totalLockedUpStake.add(spLockedStake))
+            totalLockedUpStake
         );
 
         // Amount stored in staking contract for owner
-        uint _totalBalanceInStaking = Staking(stakingAddress).totalStakedFor(msg.sender);
-        require(_totalBalanceInStaking > 0, "Stake required for claim");
+        totalBalanceInStaking = Staking(stakingAddress).totalStakedFor(msg.sender);
+        require(totalBalanceInStaking > 0, "Stake required for claim");
 
         // Amount in sp factory for claimer
-        (uint _totalBalanceInSPFactory,,,,,) = spFactory.getServiceProviderDetails(msg.sender);
-
-        // Decrease total balance by any locked up stake
-        _totalBalanceInSPFactory = _totalBalanceInSPFactory.sub(spLockedStake);
-
+        (totalBalanceInSPFactory,,,,,) = spFactory.getServiceProviderDetails(msg.sender);
         // Require active stake to claim any rewards
-        require(_totalBalanceInSPFactory > 0, "Service Provider stake required");
+        require(totalBalanceInSPFactory.sub(spLockedStake) > 0, "Service Provider stake required");
 
         // Amount in delegate manager staked to service provider
-        uint _totalBalanceOutsideStaking = (
-            _totalBalanceInSPFactory.add(spDelegateInfo[msg.sender].totalDelegatedStake)
+        totalBalanceOutsideStaking = (
+            totalBalanceInSPFactory.add(spDelegateInfo[msg.sender].totalDelegatedStake)
         );
 
-        return (_totalBalanceInStaking, _totalBalanceInSPFactory, _totalBalanceOutsideStaking);
+        require(
```

Comment:
should this be an assert?

Reply:
based on this, i don't think so - https://solidity.readthedocs.io/en/develop/control-structures.html#error-handling-assert-require-revert-and-exceptions

Reply:
"require function should be used to ensure valid conditions that cannot be detected until execution time. This includes conditions on inputs or return values from calls to external contracts"

Reply:
nvm, require is good